src/map_field.cpp

```diff
@@ -436,483 +441,481 @@ void map::process_fields_in_submap( submap *const current_submap,
-                        int num_bolts = rng( 3, 6 );
-                        for( int i = 0; i < num_bolts; i++ ) {
-                            int xdir = 0;
-                            int ydir = 0;
-                            while( xdir == 0 && ydir == 0 ) {
-                                xdir = rng( -1, 1 );
-                                ydir = rng( -1, 1 );
-                            }
-                            int dist = rng( 4, 12 );
-                            int boltx = p.x;
-                            int bolty = p.y;
-                            for( int n = 0; n < dist; n++ ) {
-                                boltx += xdir;
-                                bolty += ydir;
-                                add_field( tripoint( boltx, bolty, p.z ), fd_electricity, rng( 2, 3 ) );
-                                if( one_in( 4 ) ) {
-                                    if( xdir == 0 ) {
-                                        xdir = rng( 0, 1 ) * 2 - 1;
-                                    } else {
-                                        xdir = 0;
-                                    }
-                                }
-                                if( one_in( 4 ) ) {
-                                    if( ydir == 0 ) {
-                                        ydir = rng( 0, 1 ) * 2 - 1;
-                                    } else {
-                                        ydir = 0;
-                                    }
-                                }
-                            }
-                        }
-                    }
+static void field_processor_fd_push_items( const tripoint &p, field_entry &, field_proc_data &pd )
+{
+    map_stack items = pd.here.i_at( p );
+    for( auto pushee = items.begin(); pushee != items.end(); ) {
+        if( pushee->typeId() != itype_rock ||
+            pushee->age() < 1_turns ) {
+            pushee++;
+        } else {
+            std::vector<tripoint> valid;
+            for( const tripoint &dst : points_in_radius( p, 1 ) ) {
+                if( dst != p && pd.here.get_field( dst, fd_push_items ) ) {
+                    valid.push_back( dst );
+                }
+            }
+            if( !valid.empty() ) {
+                item tmp = *pushee;
+                tmp.set_age( 0_turns );
+                pushee = items.erase( pushee );
+                tripoint newp = random_entry( valid );
+                pd.here.add_item_or_charges( newp, tmp );
+                if( pd.player_character.pos() == newp ) {
+                    add_msg( m_bad, _( "A %s hits you!" ), tmp.tname() );
+                    const bodypart_id hit = pd.player_character.get_random_body_part();
+                    pd.player_character.deal_damage( nullptr, hit, damage_instance( damage_type::BASH, 6 ) );
+                    pd.player_character.check_dead_state();
+                }
+
+                if( npc *const n = g->critter_at<npc>(newp ) ) {
```

Comment:
```suggestion
                if( npc *const n = g->critter_at<npc>( newp ) ) {
```

Reply:
Thank you. I wonder how that happened, I even have git commit hook with astyle.